### PR TITLE
[main][2/2] Repo reorg Combine

### PR DESCRIPTION
**Pull request by mallardduck**
This will merge the `rancher/helm-project-operator` repo into this (`rancher/prometheus-federator`) repo. After which we will more easily be able to iterate on the overall projects as both the `helm-*` packages are primarily used by the `prometheus-federator` tool.

---

~A follow up to and considered blocked by: https://github.com/rancher/prometheus-federator/pull/96~

~This PR will actually combine things together and will be rebased on `mallardduck:repo-reorg/prepare` until #96 is merged - at which time, this PR's branch can be easily rebased on `main` and the PR will look smaller. Until then it will stay a draft.~

### CODE REVIEW

**Comment by mallardduck** [comment]
@joshmeranda / @jbiers - I think this is ready for review now. I plan to send a follow up PR that handles setting up release artifacts for all 3 of the command entry points. Essentially the same stuff I implemented in helm-project-operator for the first 2 when they were merged - but ported to this repo.

However since this PR is pretty large already, I don't want that work to hold up this phase of things. Especially since all 3 being merged into one repo means those artifacts aren't necessary for building `prometheus-federator` at all anymore.

File: packages/helm-locker/charts/values.yaml [hunk 1]
```diff
@@ -48,9 +48,9 @@ namespaceOverride: ""
 replicas: 1
 
 image:
-  repository: ghcr.io/rancher/helm-project-operator/helm-locker
-  tag: v0.3.0
-  pullPolicy: IfNotPresent
+  repository: ghcr.io/rancher/prometheus-federator/helm-locker
+  tag: head
```
Comment (joshmeranda) [inline packages/helm-locker/charts/values.yaml:52]: This needs to be a specific version, we can't ship charts that aren't pinned to an image version (going through the commits forgive me if you fix this in a later commit)

File: charts/helm-locker/0.0.2/values.yaml [hunk 2]
```diff
@@ -48,9 +48,9 @@ namespaceOverride: ""
 replicas: 1
 
 image:
-  repository: ghcr.io/rancher/helm-project-operator/helm-locker
-  tag: v0.3.0
-  pullPolicy: IfNotPresent
+  repository: ghcr.io/rancher/prometheus-federator/helm-locker
+  tag: head
```
Comment (joshmeranda) [inline charts/helm-locker/0.0.2/values.yaml:52]: same deal as above

File: charts/project-operator-example/0.0.0/Chart.yaml
Comment (joshmeranda) [inline charts/project-operator-example/0.0.0/Chart.yaml:1]: These examples should not be in the `charts` directory. They should stored under the `examples/`

**Review by joshmeranda — CHANGES_REQUESTED** [review]

File: packages/helm-locker/charts/values.yaml (hunk 1 above)
Comment (mallardduck) [reply packages/helm-locker/charts/values.yaml:52]: I didn't think we need to consider these published charts as they're documented as not for individual use.

Maybe I should just mark this package as `doNotRelease` like the `helm-project-operator` chart is and modify any CI that uses this chart to just pull from the `packages/` dir?

File: charts/project-operator-example/0.0.0/Chart.yaml
Comment (mallardduck) [reply charts/project-operator-example/0.0.0/Chart.yaml:1]: Yeah - think that was solved in latter commits.

**Review by jbiers — APPROVED** [review]

**Review by joshmeranda — APPROVED** [review]

**Pull request merged**